astrosat/management/commands/unloaddata.py

```diff
@@ -0,0 +1,180 @@
+
+        if self.verbosity >= 1:
+            self.stdout.write(
+                "Unloaded %d object(s) (of %d) from %d fixture(s)"
+                % (
+                    self.unloaded_object_count,
+                    self.fixture_object_count,
+                    self.fixture_count,
+                )
+            )
+
+    def unload_label(self, fixture_label):
+
+        show_progress = self.verbosity >= 3
+        for fixture_file, fixture_dir, fixture_name in self.find_fixtures(fixture_label):
+            _, ser_fmt, cmp_fmt = self.parse_name(os.path.basename(fixture_file))
+            open_method, mode = self.compression_formats[cmp_fmt]
+            fixture = open_method(fixture_file, mode)
+            try:
+                self.fixture_count += 1
+                objects_in_fixture = 0
+                unloaded_objects_in_fixture = 0
+                if self.verbosity >= 2:
+                    self.stdout.write(
+                        "Installing %s fixture '%s' from %s."
+                        % (ser_fmt, fixture_name, humanize(fixture_dir))
+                    )
+
+                objects = serializers.deserialize(
+                    ser_fmt,
+                    fixture,
+                    using=self.using,
+                    ignorenonexistent=self.ignore,
+                    handle_forward_references=True,
+                )
+
+                for obj in objects:
+                    objects_in_fixture += 1
+                    if (
+                        obj.object._meta.app_config in self.excluded_apps
+                        or type(obj.object) in self.excluded_models
+                    ):
+                        continue
+
+                    # here is the different bit...
+                    obj_class_label = obj.object._meta.label
+                    try:
+                        n_deleted, deleted_classes = obj.object.delete(using=self.using)
+                        unloaded_objects_in_fixture += n_deleted
+                        for key, val in deleted_classes.items():
+                            if key == obj_class_label:
+                                self.class_deletion_record[key] += (val - 1)
+                            else:
+                                self.class_deletion_record[key] += val
+                    except Exception as e:
+                        # various exceptions might ocurr; if it is b/c obj already doesn't exist
+                        # then that's okay, just make a note of it and move on
+                        if obj.object.pk is None:
+                            if obj_class_label in self.class_deletion_record and self.class_deletion_record[obj_class_label] > 0:
+                                self.class_deletion_record[obj_class_label] -= 1
+                                continue  # obj has been deleted in a prior loop
+                            elif self.ignore:
+                                continue  # obj wasn't in the db in the 1st place
```

Comment:
Could these `continue` not be do after the if/else block as it happens in both, would save duplication? If not, can you explain why.

Reply:
No.  

This `if/else` block could be simplified.  But putting the `continue` statement after it means it would _always_ continue if `obj.object.pk` is `None`.  

I _only_ want to continue if that is true and one of the two nested if statements is true: either I deleted it b/c of a CASCADE operation when deleting a previous object or it really is missing and I explicitly passed the `--ignorenonexistent` argument.